src/lib/progress-bar/progress-bar.ts

```diff
@@ -113,6 +131,21 @@ export class MatProgressBar extends _MatProgressBarMixinBase implements CanColor
       return {transform: `scaleX(${scale})`};
     }
   }
+
+  /** Attaches value bar element animation end listener. */
+  ngAfterViewInit() {
+    if(this.mode === 'determinate' || this.mode === 'buffer') {
+      this._valueAnimationSubscription = fromEvent(this.primaryValueBar.nativeElement, 'transitionend')
```

Comment:
This needs a couple of things:

1. Since the `transitionend` event bubbles, it means that it could fire if a transition completes for an element inside the track. We should filter out based on whether `event.target === this.primaryValueBar.nativeElement`.
2. The event should be bound outside the `NgZone` and it should re-enter the zone only when it emits.

Reply:
Couple questions on these before I make the change:
1. The event is bound to specifically div mat-progress-bar-primary element inside the template, so adding the filter target would be redundant? What am I missing here?
2. Is this more of best practice to run CSS event on NgZone? Would re-enter zone looks like this? https://github.com/angular/material2/blob/c2fc3f45b5ccdbff3ade56e0685b621688e9cb6d/src/cdk/text-field/autofill.ts#L72.

Reply:
1. That's true, but it's more of a future-proofing thing in case we decide to put something into the element.
2. It's only really necessary if we end up doing point 1, otherwise it would be redundant.

Reply:
Awesome, thanks for the explanation. I've updated the latest commit with these two points incorporated. PTAL.